### PR TITLE
FF7: Added functionality to replace worldmap rendering with an external 3d mesh

### PR DESCRIPTION
**Pull request by CosmosXIII**
## Summary

This PR adds functionality to replace worldmap terrain rendering with an external 3d mesh. This feature requires the mesh data to be provided with a separate mod (an iro for 7thHeaven).

### Motivation

To make people (me included) happy.

### ACKs

- [x] I have updated the [Changelog.md](https://github.com/julianxhokaxhiu/FFNx/blob/master/Changelog.md) file
- [x] I did test my code on FF7
- [ ] I did test my code on FF8

### CODE REVIEW

**Review by julianxhokaxhiu — DISMISSED** [review]
LGTM - A bit massive PR to review but overall I wasn't able to spot anything critical. Let me know if it good to be merged. If that's the case.

File: misc/FFNx.lighting.vert [hunk 1]
```diff
@@ -44,6 +45,9 @@ uniform vec4 gameScriptedLightColor;
 #define isFBTexture VSFlags.z > 0.0
 #define isNotTexture VSFlags.w == 0.0
 
+#define isApplySphericalWorld WMFlags.x > 0.0
+#define sphericaWorldRadiusScale WMFlags.x
```
Comment (KuraudoXS) [inline misc/FFNx.lighting.vert:49]: spherical?

**Review by julianxhokaxhiu — APPROVED** [review]

**Comment by julianxhokaxhiu** [comment]
Thank you!

File: misc/FFNx.lighting.vert (hunk 1 above)
[the remark this answers is not in the document]
Comment (julianxhokaxhiu) [reply misc/FFNx.lighting.vert:49]: Feel free to open a MR, I'll be more than happy to review if you find any typo around :) Thank you!

**Pull request merged**